OptimizelySDK/Optimizely.cs

```diff
@@ -836,13 +846,63 @@ public OptimizelyJSON GetFeatureVariableJSON(string featureKey, string variableK
                 reasonsToReport.ToArray());
         }
 
+        internal Dictionary<string, OptimizelyDecision> DecideAll(OptimizelyUserContext user,
+                                              OptimizelyDecideOption[] options)
+        {
+            var decisionMap = new Dictionary<string, OptimizelyDecision>();
+
+            var projectConfig = ProjectConfigManager?.GetConfig();
+            if (projectConfig == null)
+            {
+                Logger.Log(LogLevel.ERROR, "Optimizely instance is not valid, failing isFeatureEnabled call.");
+                return decisionMap;
+            }
+
+            var allFlags = projectConfig.FeatureFlags;
+            var allFlagKeys = allFlags.Select(v => v.Key).ToArray<string>();
+            
+            return DecideForKeys(user, allFlagKeys, options);
+        }
+
+
+        internal Dictionary<string, OptimizelyDecision> DecideForKeys(OptimizelyUserContext user,
+                                                      string[] keys,
+                                                      OptimizelyDecideOption[] options)
+        {
+            var decisionDictionary = new Dictionary<string, OptimizelyDecision>();
+
+            var projectConfig = ProjectConfigManager?.GetConfig();
+            if (projectConfig == null)
+            {
+                Logger.Log(LogLevel.ERROR, "Optimizely instance is not valid, failing isFeatureEnabled call.");
+                return decisionDictionary;
+            }
+
+            if (keys.Length == 0)
+            { 
+                return decisionDictionary;
+            }
+
+            var allOptions = GetAllOptions(options);
+
+            foreach (string key in keys)
+            {
+                var decision = Decide(user, key, options);
+                if (!allOptions.Contains(OptimizelyDecideOption.ENABLED_FLAGS_ONLY) || decision.Enabled)
+                {
+                    decisionDictionary.Add(key, decision);
+                }
+            }
+
+            return decisionDictionary;
+        }
         private OptimizelyDecideOption[] GetAllOptions(OptimizelyDecideOption[] options)
         {
             OptimizelyDecideOption[] copiedOptions = new OptimizelyDecideOption[DefaultDecideOptions.Length];
             Array.Copy(DefaultDecideOptions, copiedOptions, DefaultDecideOptions.Length);
             if (options != null)
             {
-                copiedOptions.Concat(options);
+                copiedOptions = copiedOptions.Concat(options).Concat(DefaultDecideOptions).ToArray();
```

Comment:
Can you please try, it's just an improvement. 
`return = options.Union(DefaultDecideOptions).ToArray();`